### PR TITLE
rosa: 1.2.46 -> 1.2.48

### PR DESCRIPTION
**Pull request by jfchevrette**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/ro/rosa/package.nix [hunk 1]
```diff
@@ -22,10 +22,9 @@ buildGoModule rec {
   '';
 
   preCheck = ''
-    # Workaround for cmd/list/rhRegion/cmd_test.go:39
-    #   Failed to get OCM regions: Can't retrieve shards: Get "https://api.stage.openshift.com/static/ocm-shards.json": dial tcp: lookup api.stage.openshift.com on [::1]:53: read udp [::1]:55793->[::1]:53: read: connection refused
-    substituteInPlace "cmd/list/rhRegion/cmd_test.go" \
-      --replace-fail "TestRhRegionCommand" "SkipRhRegionCommand"
+    # remove tests that require network access
+    rm cmd/list/rhRegion/cmd_test.go
+    rm cmd/describe/cluster/cmd_test.go
```
Comment (getchoo) [inline pkgs/by-name/ro/rosa/package.nix:27]: These should be skipped in the way described by the [Go chapter of the manual](https://nixos.org/manual/nixpkgs/stable/#ssec-skip-go-tests)

**Review by getchoo — CHANGES_REQUESTED** [review]

File: pkgs/by-name/ro/rosa/package.nix (hunk 1 above)
Comment (jfchevrette) [reply pkgs/by-name/ro/rosa/package.nix]: Thank you for taking the time to point this out. I've reviewed the docs you linked as well as the [buildGoModule source](https://github.com/NixOS/nixpkgs/blob/master/pkgs/build-support/go/module.nix) and I think the refactored commit is better, including the change to use `excludedPackages` to exclude the `tests/e2e` test suite

File: pkgs/by-name/ro/rosa/package.nix [hunk 2]
```diff
@@ -2,31 +2,25 @@
...
+  excludedPackages = [ "tests/e2e" ];
 
-  preCheck = ''
-    # Workaround for cmd/list/rhRegion/cmd_test.go:39
-    #   Failed to get OCM regions: Can't retrieve shards: Get "https://api.stage.openshift.com/static/ocm-shards.json": dial tcp: lookup api.stage.openshift.com on [::1]:53: read udp [::1]:55793->[::1]:53: read: connection refused
-    substituteInPlace "cmd/list/rhRegion/cmd_test.go" \
-      --replace-fail "TestRhRegionCommand" "SkipRhRegionCommand"
-  '';
+  # skip tests that require network access
+  checkFlags = [ "-skip=^TestCluster$" ];
```
Comment (getchoo) [inline pkgs/by-name/ro/rosa/package.nix:23]:
```suggestion
  checkFlags =
    let
      skippedTests = [
        "TestRhRegionCommand"
        "TestCluster"
      ];
    in
    [ "-skip=^${lib.concatStringsSep "$|^" skippedTests}$" ];
```

The test previously patched out [still fails](https://logs.ofborg.org/?key=nixos/nixpkgs.359440&attempt_id=1ba2ef0e-dae8-496d-966e-d28e394ab8b9) here. Adding this fixed building on x86_64-linux

**Review by getchoo — CHANGES_REQUESTED** [review]

File: pkgs/by-name/ro/rosa/package.nix (hunk 2 above)
Comment (jfchevrette) [reply pkgs/by-name/ro/rosa/package.nix]: I didn't have a proper linux/nixos environment to test in and that test does not fail on darwin (with sandboxing). In any case, I have re-added that test to the list and now I have a linux environment to test builds in. Thank you!

**Comment by getchoo** [comment]
@ofborg build rosa

File: pkgs/by-name/ro/rosa/package.nix [hunk 3]
```diff
@@ -2,31 +2,32 @@
...
+  # skip e2e tests package
+  excludedPackages = [ "tests/e2e" ];
+
+  # skip tests that require network access
+  checkFlags =
+    let
+      skippedTests = [
+        "TestCluster"
+        "TestRhRegionCommand"
+      ];
```
Comment (getchoo) [inline pkgs/by-name/ro/rosa/package.nix:28]:
```suggestion
      ]
      ++ lib.optional stdenv.hostPlatform.isDarwin "TestCache";
```
Seems there are some [permission issues](https://logs.ofborg.org/?key=nixos/nixpkgs.359440&attempt_id=62589cca-19f6-4366-86c4-3f93476a1e46) with temporary files on Darwin. Pretty sure this is fine

**Review by getchoo — CHANGES_REQUESTED** [review]

File: pkgs/by-name/ro/rosa/package.nix (hunk 3 above)
Comment (jfchevrette) [reply pkgs/by-name/ro/rosa/package.nix:28]: Strangely enough, this test did not fail on aarch64-darwin (with sandbox). I'll add this one too. Again thank you for your help with this one!

**Comment by jfchevrette** [comment]
@ofborg build rosa

**Review by getchoo — APPROVED** [review]
Nice work :)

## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 359440`

---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>rosa</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>rosa</li>
  </ul>
</details>

**Pull request merged**